pinot-core/src/test/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManagerTest.java

```diff
@@ -137,11 +142,13 @@ private FakeLLRealtimeSegmentDataManager createFakeSegmentManager()
     InstanceZKMetadata instanceZKMetadata = new InstanceZKMetadata();
     RealtimeTableDataManager tableDataManager = createTableDataManager();
     String resourceDir = _segmentDir;
+    LLCSegmentName llcSegmentName = new LLCSegmentName(_segmentNameStr);
+    _partitionIdToSemaphoreMap.putIfAbsent(_partitionId, new Semaphore(1));
```

Comment:
Should you add this in the test setup method?

Reply:
That's been added.